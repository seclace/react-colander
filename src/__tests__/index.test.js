import React, { Component } from 'react'
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { colander } from './../index';

describe('React', () => {

  const props = { fetching: false, user: { id: 1 } };

  class Foo extends Component {
    render() {
      return <div className="foo" />
    }
  }

  class Preloader extends Component {
    render() {
      return <div className="preloader" />
    }
  }

  class UnauthMessage extends Component {
    render() {
      return <div className="unauth-message" />
    }
  }

  function preload(props) {
    if (props.fetching) return <Preloader />;
  }

  function unauth(props) {
    if (!props.user) return <UnauthMessage />;
  }

  describe('colander', () => {

    const FooComposer = colander(Foo, [preload, unauth]);
    const wrapper = mount( <FooComposer {...props} /> );

    it('should render Foo wrapped to Colander', () => {
      wrapper.setProps(props);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Preloader instead Foo', () => {
      const props = { fetching: true, user: null };
      wrapper.setProps(props);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render UnauthMessage instead Foo', () => {
      const props = { fetching: false, user: null };
      wrapper.setProps(props);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

});
