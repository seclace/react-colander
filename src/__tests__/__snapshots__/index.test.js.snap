// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React colander should render Foo wrapped to Colander 1`] = `
<ComposerWrapper
  fetching={false}
  user={
    Object {
      "id": 1,
    }
  }
>
  <Colander
    colanders={
      Array [
        [Function],
        [Function],
      ]
    }
    props={
      Object {
        "fetching": false,
        "user": Object {
          "id": 1,
        },
      }
    }
  >
    <Foo
      fetching={false}
      user={
        Object {
          "id": 1,
        }
      }
    >
      <div
        className="foo"
      />
    </Foo>
  </Colander>
</ComposerWrapper>
`;

exports[`React colander should render Preloader instead Foo 1`] = `
<ComposerWrapper
  fetching={true}
  user={null}
>
  <Colander
    colanders={
      Array [
        [Function],
        [Function],
      ]
    }
    props={
      Object {
        "fetching": true,
        "user": null,
      }
    }
  >
    <Preloader>
      <div
        className="preloader"
      />
    </Preloader>
  </Colander>
</ComposerWrapper>
`;

exports[`React colander should render UnauthMessage instead Foo 1`] = `
<ComposerWrapper
  fetching={false}
  user={null}
>
  <Colander
    colanders={
      Array [
        [Function],
        [Function],
      ]
    }
    props={
      Object {
        "fetching": false,
        "user": null,
      }
    }
  >
    <UnauthMessage>
      <div
        className="unauth-message"
      />
    </UnauthMessage>
  </Colander>
</ComposerWrapper>
`;
